src/MainWindow/main_window.cpp

```diff
@@ -160,6 +160,18 @@ void MainWindow::CloseOpenedTabs() {
   }
 }
 
+void MainWindow::closeEvent(QCloseEvent* event) {
+  QMessageBox::StandardButton closeBtn = QMessageBox::warning(
+      this, "Warning", tr("Do you really want to close the project?\n"),
+      QMessageBox::Cancel | QMessageBox::Yes);
+
+  if (closeBtn != QMessageBox::Yes) {
```

Comment:
why you invert logic? **If (closeBtn == QMessageBox::Yes)** looks more straight

Reply:
Fixed.